samples/Mvc.ControllerForMultipleAreasSample/MultipleAreasControllerConvention.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+
+namespace ControllerForMultipleAreasSample
+{
+    public class MultipleAreasControllerConvention : IApplicationModelConvention
+    {
+        public MultipleAreasControllerConvention()
+        {
+            ControllerModels = new List<ControllerModel>();
+        }
+
+        public IList<ControllerModel> ControllerModels { get; set; }
+
+        public void Apply(ApplicationModel application)
+        {
+            foreach (var controller in application.Controllers)
+            {
+                var areas = controller.ControllerType.GetCustomAttributes<MultipleAreasAttribute>().FirstOrDefault();
+                var areaNames = areas?.AreaNames;
+
+                for (var i = 0; i < areaNames?.Length; i++)
```

Comment:
a more common pattern for this is to just start the loop at 1. So move L28 outside of the loop, and then simplify the body to just L33-35